From: Stanislav Malyshev <stas@php.net>
Date: Tue Aug 16 16:03:44 2016 -0700
Subject: [PATCH] Fix bug #72850 - integer overflow in uuencode

5.4 backport: https://github.com/shivammathur/php-src-backports/commit/3450ec4fea03871a598b225efa6f993447286191
PLD: with the php_uudecode() hunk of upstream 9d7ca3857b, the 7.0 commit of this fix; the 5.4 branch left it out
---
diff --git a/ext/standard/uuencode.c b/ext/standard/uuencode.c
--- a/ext/standard/uuencode.c
+++ b/ext/standard/uuencode.c
@@ -168,7 +168,7 @@ PHPAPI int php_uudecode(char *src, int src_len, char **dest) /* {{{ */
 		s++;
 	}
 
-	if ((len = total_len > (p - *dest))) {
+	if ((len = total_len) > (p - *dest)) {
 		*p++ = PHP_UU_DEC(*s) << 2 | PHP_UU_DEC(*(s + 1)) >> 4;
 		if (len > 1) {
 			*p++ = PHP_UU_DEC(*(s + 1)) << 4 | PHP_UU_DEC(*(s + 2)) >> 2;
@@ -200,6 +200,11 @@ PHP_FUNCTION(convert_uuencode)
 	}
 
 	dst_len = php_uuencode(src, src_len, &dst);
+	if (dst_len < 0) {
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "String too long, max length is %d", INT_MAX);
+		efree(dst);
+		RETURN_FALSE;
+	}
 
 	RETURN_STRINGL(dst, dst_len, 0);
 }
